drivers/spi/spi_cc13xx_cc26xx.c

```diff
@@ -0,0 +1,298 @@
+/*
+ * Copyright (c) 2019 Brett Witherspoon
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <spi.h>
+#include <sys_clock.h>
+
+#define LOG_LEVEL CONFIG_SPI_LOG_LEVEL
+#include <logging/log.h>
+LOG_MODULE_REGISTER(spi_cc13xx_cc26xx);
+
+#include <driverlib/prcm.h>
+#include <driverlib/ssi.h>
+#include <driverlib/ioc.h>
+
+#include "spi_context.h"
+
+struct spi_cc13xx_cc26xx_config {
+	u32_t base;
+	u32_t sck_pin;
+	u32_t mosi_pin;
+	u32_t miso_pin;
+	u32_t cs_pin;
+};
+
+struct spi_cc13xx_cc26xx_data {
+	struct spi_context ctx;
+};
+
+static inline struct spi_cc13xx_cc26xx_data *get_dev_data(struct device *dev)
+{
+	return dev->driver_data;
+}
+
+static inline const struct spi_cc13xx_cc26xx_config *
+get_dev_config(struct device *dev)
+{
+	return dev->config->config_info;
+}
+
+static int spi_cc13xx_cc26xx_configure(struct device *dev,
+				       const struct spi_config *config)
+{
+	const struct spi_cc13xx_cc26xx_config *cfg = get_dev_config(dev);
+	struct spi_context *ctx = &get_dev_data(dev)->ctx;
+	u32_t prot;
+
+	if (spi_context_configured(ctx, config)) {
+		return 0;
+	}
+
+	if (SPI_OP_MODE_GET(config->operation) != SPI_OP_MODE_MASTER) {
+		LOG_ERR("Slave mode is not supported");
+		return -EINVAL;
```

Comment:
I too am interested in what others have to say. I remember trying to decide between -ENOTSUP and -EINVAL here because I was trying to do the same as other drivers. I have found that the API documentation sometimes specifies what should be returned. For example, the [I2C](https://docs.zephyrproject.org/latest/reference/peripherals/i2c.htmll) API documentation says return -EIO on any error, but I am not sure all drivers really do this. Unfortunately, the [SPI](https://docs.zephyrproject.org/latest/reference/peripherals/spi.html) API documentation just says a negative errno.

Reply:
go for -ENOTSUP